### DISCUSSION TITLE
Missing some system fonts compare with the System.Drawing

### DISCUSSION THREAD

**Discussion by linch90** [SixLabors/Fonts#305] [Q&A] [answered]
Hi. I use SystemFonts.CreateFont() method to access font in "C:\Windows\Fonts"(As my development envionment is win11).
But it throws exception that says: `SixLabors.Fonts.FontFamilyNotFoundException : The "宋体" font family could not be found in the following directory: C:\WINDOWS\Fonts`

So I create net6 console project, to compare the system fonts getting from the System.Drawing.Common and SixLabors.Fonts
```csharp
using System.Drawing.Text;
using SixLabors.Fonts;

var fontCollectionFromSystemDrawing = new InstalledFontCollection();
var fontFamiliesFromSystemDrawing = fontCollectionFromSystemDrawing.Families
    .Select(ff => ff.Name).ToArray();  // string[278]

var fontFamiliesFromSixLabors = SystemFonts.Families
    .Select(ff => ff.Name)
    .ToArray(); // string[236]

var missingFontFamiliesOfSixLabors =
    fontFamiliesFromSystemDrawing.Except(fontFamiliesFromSixLabors).ToList(); // Count = 71
```
You can see a lot of chinese fonts given by the System.Drawing:
![image](https://user-images.githubusercontent.com/1113475/197376753-34c149ae-ac87-43a4-ac25-8099dcef8ae3.png)

but no chinese fonts given by the Sixlabors.Fonts package.
![image](https://user-images.githubusercontent.com/1113475/197376833-d7ed000d-0985-495c-8120-66002d0827dd.png)

**Comment by linch90** [comment]
![image](https://user-images.githubusercontent.com/1113475/197376924-057e6794-0494-41c6-8c13-5b079602c406.png)
This is the screenshots for some fonts in my "C:\Windows\Fonts" folder, it contains the chinese fonts as the System.Drawing display.

**Reply by JimBobSquarePants** [reply]
Is it possible to explicitly load any of those missing fonts by using the exact path? We currently ignore issues when trying to load system fonts so I'm wondering whether there is something failing that prevents us loading them 

https://github.com/SixLabors/Fonts/blob/380b6680fab5153edd6dc435c743150bd5a97598/src/SixLabors.Fonts/SystemFontCollection.cs#L153-L169

**Reply by linch90** [reply]
Hi, after some coding, I found it's not fonts missing, but the name of some fonts in SixLabors is diff from System.Drawing.
Model:
```csharp
class FontCollectionPair
{
    public FontCollection SixLabors { get; set; }

    public PrivateFontCollection SystemDrawing { get; set; }

    public FontCollectionPair(FontCollection sixLabors, PrivateFontCollection systemDrawing)
    {
        SixLabors = sixLabors;
        SystemDrawing = systemDrawing;
    }

    public override string ToString()
    {
        return $"SixLabors: {string.Join(',', SixLabors.Families.Select(f => f.Name))} | System.Drawing: {string.Join(',', SystemDrawing.Families.Select(f => f.Name))}";
    }
}
```
Logic:
```csharp
using System.Drawing.Text;
using FontCollection = SixLabors.Fonts.FontCollection;

const string fontsFolder = @"C:\Windows\Fonts";
var files = Directory.EnumerateFiles(fontsFolder);

var fontCollectionPathPairMap = new Dictionary<string, FontCollectionPair>();

foreach (var filePath in files)
{
    if (filePath.EndsWith(".ttc", StringComparison.OrdinalIgnoreCase))
    {
        var sixLabors = new FontCollection();
        sixLabors.AddCollection(filePath);
        var systemDrawing = new PrivateFontCollection();
        systemDrawing.AddFontFile(filePath);
        if (sixLabors.Families.Select(f => f.Name).OrderBy(x => x).SequenceEqual(
                systemDrawing.Families.Select(f => f.Name).OrderBy(x => x)
            ))
        {
            continue; // Skip if the font names is sequence equal
        }
        fontCollectionPathPairMap[filePath] = new FontCollectionPair(sixLabors, systemDrawing);
    }
    else if (filePath.EndsWith(".ttf", StringComparison.OrdinalIgnoreCase))
    {
        var sixLabors = new FontCollection();
        sixLabors.Add(filePath);
        var systemDrawing = new PrivateFontCollection();
        systemDrawing.AddFontFile(filePath);
        if (sixLabors.Families.Select(f => f.Name).OrderBy(x => x).SequenceEqual(
                systemDrawing.Families.Select(f => f.Name).OrderBy(x => x)
            ))
        {
            continue; // Skip if the font names is sequence equal
        }
        fontCollectionPathPairMap[filePath] = new FontCollectionPair(sixLabors, systemDrawing);
    }
}
```
As the result shows: there are 34 fonts having diff names, following screenshots shows some samples:
![image](https://user-images.githubusercontent.com/1113475/197391627-5ccbd87b-12c1-444d-9711-175e564998f8.png).

As you can see, the System.Drawing give the chinese characters name for some fonts, when the SixLabors always give the english characters name.
So when I use the given chinese characters family name to create font in SixLabors, the exception will throw.

**Reply by JimBobSquarePants** [reply]
Ah. That is because we use the invariant culture when adding the font. I'm not sure it's wise to use the current culture as we might have no guarantee that cultures are configured properly.

**Reply by linch90** [reply]
Hi, so I change the codes 
`sixLabors.AddCollection(filePath);` to `sixLabors.AddCollection(filePath, CultureInfo.CurrentCulture);`
`sixLabors.Add(filePath);` to `sixLabors.Add(filePath, CultureInfo.CurrentCulture);`
The CurrentCulture is 'zh-CN' which is chinese culture. but still the same result.
It seems the current culture do not work properly.

**Comment by linch90** [comment, marked as answer]
Hi, @JimBobSquarePants , I see the source code of SixLabors.Fonts, the SystemFonts.CreateFont() is invariant by default, but it has provided overloads method that has a CultureInfo parameter for create current culture font, thus this is not a problem or bug, just my unfamiliar with this Fonts package.
Anyway, thanks for taking the time to answer my question.
Best regards.

**Reply by JimBobSquarePants** [reply]
No worries. It's great to see someone exploring the API!
